parlai/core/torch_generator_agent.py

```diff
@@ -1043,8 +1049,7 @@ def select_paths(self, logprobs, prior_scores):
         hyp_ids = torch.arange(logprobs.size(0)).to(logprobs.device)
         tok_ids = indices[hyp_ids, choices]
         scores = values[hyp_ids, choices]
-        best_scores = prior_scores.expand_as(scores) + scores
-        return (hyp_ids, tok_ids, best_scores)
+        return (hyp_ids, tok_ids, scores)
```

Comment:
Why are you changing this? The contract of tree search is the logprob of the full utterance is contained in the third return value.

Reply:
I changed this to avoid having -inf scores here after the first EOS was sampled. But actually thats fine if all finalized hypos after the first one will have -inf score, I will change this back. Then only the first sampled EOS will make a non -inf finalized hypo per each beam id